Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/common/AzureStorageUri.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) Microsoft Corporation
+ *
+ * All rights reserved.
+ *
+ * MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+ * documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+ * to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of
+ * the Software.
+ *
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+ * THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.microsoft.azure.hdinsight.common;
+
+import java.net.URI;
+import java.net.URL;
+
+/**
+ * A class for Azure Storage related URIs
+ */
+abstract public class AzureStorageUri {
+    private final URI rawUri;
+
+    protected AzureStorageUri(URI rawUri) {
+        this.rawUri = rawUri;
+    }
+
+    /**
+     * Getter of raw URI
+     *
+     * @return the raw URI
+     */
+    public URI getRawUri() {
+        return rawUri;
+    }
+
+    /**
+     * Get URI with specified scheme, not limited to HTTP or HTTPS
```

Comment:
When you say `not limited to HTTP or HTTPS`, do you mean `getUri()` can return URIs not only starting with `HTTP` or `HTTPS` but also other schemes?

Reply:
Yes, such as wasbs://, adl://, abfs:// and so on.

Reply:
I suppose `getUri` should return URIs not starting with `http` because we need to differenciate it with `getUrl`.